server/controllers/authController.api.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const db = require('../config/db.config.js');
+const User = db.user;
+const jwt = require('jsonwebtoken');
+const passport = require('passport')
+const bcrypt = require('bcrypt');
+const saltRounds = 12;
+const JWTSecret = process.env.JWT_SECRET || 'some secret';
+
+
+exports.register = (req, res) => {
+    const username = req.body.username;
+    const email = req.body.email;
+    const password = req.body.password;
+
+    const newUser = new User({
+      username,
+      email,
+      password
```

Comment:
I don't think the password is being encrypted when it is saved? There are two ways I know how to do it, either in the User model using a virtual field for password with setter and getter and also adding an authenticate and encryptPassword method onto the userSchema. The other way is you can do it here in the controller. Before you save this user, just use bcrypt and generate the salt then hash it. 
Personally I like doing it in the User model because it saves lines in the controller. However, I'm open to any way you decide to choose!

Reply:
Yep. I'm going to do the encryption and validation in the User model, which I noted above I haven't pushed yet. I have the bcrypt methods written out but haven't hooked up to the db yet.

Reply:
Just pushed out a rough version of the user schema -- it's only the bcrypt boilerplate at this point.